statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -1113,6 +1113,198 @@ def zvalues(self):
+            Whether or not to compare against the asymptotic distribution
+            (chi-squared) or the approximate small-sample distribution (F).
+            Default is False (i.e. default is to compare against an F
+            distribution).
+
+        Notes
+        -----
+        The null hypothesis is of no heteroskedasticity. That means different
+        things depending on which alternative is selected:
+
+        - Increasing: Null hypothesis is that the variance is not increasing
+          throughout the sample; that the sum-of-squares in the later
+          subsample is *not* greater than the sum-of-squares in the earlier
+          subsample.
+        - Decreasing: Null hypothesis is that the variance is not decreasing
+          throughout the sample; that the sum-of-squares in the earlier
+          subsample is *not* greater than the sum-of-squares in the later
+          subsample.
+        - Two-sided: Null hypothesis is that the variance is not changing
+          throughout the sample. Both that the sum-of-squares in the earlier
+          subsample is not greater than the sum-of-squares in the later
+          subsample *and* that the sum-of-squares in the later subsample is
+          not greater than the sum-of-squares in the earlier subsample.
+
+        For :math:`h = [T/3]`, the test statistic is:
+
+        .. math::
+
+            H(h) = \sum_{t=T-h+1}^T  \tilde v_t^2
+            \left / \sum_{t=d+1}^{d+1+h} \tilde v_t^2 \right .
+
+        where :math:`d` is the number of periods in which the loglikelihood was
+        burned in the parent model (usually corresponding to diffuse
+        initialization).
+
+        This statistic can be tested against an :math:`F(h,h)` distribution.
+        Alternatively, :math:`h H(h)` is asymptotically distributed according
+        to :math:`\chi_h^2`; this second test can be applied by passing
+        `asymptotic=True` as an argument.
+
+        See section 5.4 of [1]_ for the above formula and discussion, as well
+        as additional details.
+
+        TODO
+
+        - Allow specification of :math:`h`
+
+        Returns
+        -------
+        output : array
+            An array with `(test_statistic, pvalue)` for each endogenous
+            variable. The array is then sized `(k_endog, 2)`. If the method is
+            called as `het = res.test_heteroskedasticity()`, then `het[0]` is
+            an array of size 2 corresponding to the first endogenous variable,
+            where `het[0][0]` is the test statistic, and `het[0][1]` is the
+            p-value.
+
+        References
+        ----------
+        .. [1] Harvey, Andrew C. 1990.
+           Forecasting, Structural Time Series Models and the Kalman Filter.
+           Cambridge University Press.
+
+        """
+        # Store some values
+        squared_resid = self.filter_results.standardized_forecasts_error**2
+        d = self.loglikelihood_burn
+        h = np.round((self.nobs - d) / 3)
```

Comment:
I guess this needs an `int` in front or we get a deprecation warning with numpy indexing below.